packages/wallet-adapter-core/src/WalletCore.ts

```diff
@@ -98,6 +138,117 @@ export class WalletCore extends EventEmitter<WalletCoreEvents> {
     });
   }
 
+  private fetchAptosWallets() {
+    let { aptosWallets, on } = getAptosWallets();
+    this.setWallets(aptosWallets);
+
+    if (typeof window === "undefined") return;
+    // Adds an event listener for new wallets that get registered after the dapp has been loaded,
+    // receiving an unsubscribe function, which it can later use to remove the listener
+    const that = this;
+    const removeRegisterListener = on("register", function () {
+      let { aptosWallets } = getAptosWallets();
+      that.setWallets(aptosWallets);
+    });
+
+    const removeUnregisterListener = on("unregister", function () {
+      let { aptosWallets } = getAptosWallets();
+      that.setWallets(aptosWallets);
+    });
```

Comment:
Maybe there's no need to worry about it at this point, but we need to at least give users a way to unregister the listeners.

What happens if you instantiate a WalletCore, then destroy it and instantiate a new one?
Not sure when that would happen, but it could.
Like when using react's `StrictMode`

Reply:
No action item for this PR necessarily, but wanted to surface this jic